Removed ingesters blocks transfer support (#2996)
cortexproject/cortex: docs/guides/ingesters-rolling-updates.md

During a rolling update, the leaving ingester closes the open TSDBs, synchronize the data to disk (`fsync`) and releases the disk resources.
The new ingester, which is expected to reuse the same disk of the leaving one, will replay the TSDB WAL on startup in order to load back in memory the time series that have not been compacted into a block yet.

_The blocks storage doesn't support the series [hand-over](#chunks-storage-with-wal-disabled-hand-over)._

## Chunks storage

The Cortex chunks storage optionally supports a write-ahead log (WAL).
The rolling update procedure for a Cortex cluster running the chunks storage depends whether the WAL is enabled or not.

### Chunks storage with WAL enabled

Similarly to the blocks storage, when Cortex is running the chunks storage with WAL enabled, it requires ingesters to run with a persistent disk where the WAL is stored (eg. a StatefulSet when deployed on Kubernetes).

During a rolling update, the leaving ingester closes the WAL, synchronize the data to disk (`fsync`) and releases the disk resources.

> The new ingester, which is expected to reuse the same disk of the leaving one, will replay the WAL on startup in order to load back in memory the time series that have not been flushed to the storage yet.

Reviewer: It actually loads all data into memory, even flushed samples. WAL doesn't preserve "flushed" flag, so data will be possibly flushed twice.

Author: Right. I'm not sure we should mention the "data will be possibly flushed twice" here, given this page is focused on rolling updates. Maybe it's something we should improve in the "Ingesters with WAL" page. WDYT?

In the meanwhile, I've rephrased it inhttps://github.com/cortexproject/cortex/pull/2996/commits/caa7dd453b918cd0674cbc52d999490e3ceebe0b.

Reviewer: Nit: In that change you have introduce "checkpoint" term, with no explanation. I think we can be bit more vague and just say that ingester reloads data from WAL.

Author: Agree, done.

_For more information about the WAL, please refer to [Ingesters with WAL](../production/ingesters-with-wal.md)._

### Chunks storage with WAL disabled (hand-over)

When Cortex is running the chunks storage with WAL disabled, Cortex supports on-the-fly series hand-over between a leaving ingester and a joining one.

The hand-over is based on the ingesters state stored in the ring. Each ingester could be in one of the following **states**:

- `PENDING`
- `JOINING`
- `ACTIVE`
- `LEAVING`

On startup, an ingester goes into the **`PENDING`** state.
In this state, the ingester is waiting for a hand-over from another ingester that is `LEAVING`.
If no hand-over occurs within the configured timeout period ("auto-join timeout", configurable via `-ingester.join-after` option), the ingester will join the ring with a new set of random tokens (eg. during a scale up) and will switch its state to `ACTIVE`.

> A running ingester, in the **`ACTIVE`** state, is notified to shutdown via the Unix signal `SIGINT`.
> On receipt of this signal, the ingester switches to `LEAVING`: in this state it cannot receive write requests anymore, while it could still receive read requests for series it has in memory.
> 

Reviewer: ```suggestion
When a running ingester in the **`ACTIVE`** state is notified to shutdown via `SIGINT` or `SIGTERM` Unix signal, the ingester switches to `LEAVING` state. In this state it cannot receive write requests anymore, but it can still receive read requests for series it has in memory.
```
